api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/DistrictService.java

```diff
@@ -28,6 +32,8 @@ public class DistrictService {
     DistrictRedisRepository districtRedisRepository;
     @Autowired
     DistrictTransformer districtTransformer;
+    @Autowired
+    RedisTemplate<String, String> redisTemplate;
 
     public List<District> getDistrictsFromInstituteApi() {
         try {
```

Comment:
Create a webclient and qualifier for grad-edx-client like the pattern established [here](https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/main/api/src/main/java/ca/bc/gov/educ/api/trax/config/GradTraxConfig.java#L56) and used [here](https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/main/api/src/main/java/ca/bc/gov/educ/api/trax/util/CommonSchoolCache.java#L29) and configured [here](https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/main/api/src/main/resources/application.yaml#L47). Also we should add some resilience to the call in case of a 5xx response (like 503 due to pod restarts, etc.) There is an example of how you can add simple resilience to REST calls [here](https://github.com/bcgov/EDUC-GRAD-GRADUATION-API/blob/main/api/src/main/java/ca/bc/gov/educ/api/graduation/service/RESTService.java#L73)

Reply:
WebClient Fixed:
https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/c79bdcc841ed4529314fd9ef0d3bd821fb053b15/api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/CodeService.java#L31

Reply:
Resilience added:
https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/c79bdcc841ed4529314fd9ef0d3bd821fb053b15/api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/CodeService.java#L48